dynamite_nsm/services/kibana/package/manager.py

```diff
@@ -0,0 +1,784 @@
+
+        Raises:
+            ValueError: If response from ES is nonsuccess (not 200 series status)
+
+        Returns:
+            result: dict
+        """
+        result = requests.get(f"{Package.es_url}/{Package.package_index_name}/_search/",
+                      json=query,
+                      verify=False,
+                      auth=Package.auth)
+        if result.status_code not in range(200,299):
+            raise ValueError(f"Failed to fetch package data")
+        num_returned = result.json().get('hits', {
+            "total" : {
+                "value" : 0,
+            },
+            "hits" : []
+        }).get('total').get('value')
+
+        if not num_returned:
+             return {}
+        return result.json()
+
+    def reload_installed_objects(self) -> list:
+        """Fetches Package information from ES
+
+        Performs a query based on the existing package manifest
+        and loads all installed objects 
+        into self._installed_objects
+        
+
+        Returns:
+            list[InstalledObjects]: List of installed Objects
+        """
+        # should we instead perform inner hits query on nested object to get objects for current slug?
+        query = {
+            "query": {
+                "match":{
+                    "manifest.slug":{
+                        "query": self.slug,
+                        "minimum_should_match": 1
+                    }
+                }
+            }
+        }
+        
+        result = Package.es_search(query)
+        if not result:
+            return []
+        packagesdata = [r['_source'] for r in result['hits']['hits']]
+        pkg = packagesdata[0]
+        instobjs = [InstalledObject.from_kwargs(**iobj) for iobj in pkg.get('installed_objects')]
+        self._installed_objects = instobjs
+        return self._installed_objects
+        
+
+        
+    @property
+    def installed_objects(self):
+        if not self._installed_objects:
+            self.reload_installed_objects()
+        return self._installed_objects
+
+    def _check_index_exists(self) -> bool:
+        """Checks if the dynamite packages index exists in ES
+
+        Returns:
+            bool: success
+        """
+        res = requests.head(f"{self.es_url}/{self.package_index_name}", verify=False, auth=self.auth)
+        return res.status_code == 200
+
+    @staticmethod
+    def load_from_archive(package_path):
+        """[summary]
+
+        Args:
+            package_path (str): path to the archive on disk
+
+        Returns:
+            Package: a Package instance from the loaded package file.
+        """
+        tar = tarfile.open(package_path)
+        try:
+            manifest = tar.extractfile('manifest.json')
+            manifest = PackageManifest(manifest.read().decode('utf8'))
+            package = Package(manifest)
+            return package
+        except KeyError:
+            print("Package must contain a manifest.json")
```

Comment:
Please log the error then raise the exception. Do not use print except maybe for user input prompts.